core/src/main/java/org/elasticsearch/search/aggregations/metrics/stats/extended/InternalExtendedStats.java

```diff
@@ -91,6 +91,20 @@ public double value(String name) {
     }
 
     @Override
+    public Object getProperty(List<String> path) {
+        if (path.size() == 2 && "std_deviation_bounds".equals(path.get(0))) {
+            String bound = path.get(1);
+            if ("lower".equals(bound)) {
+                return getStdDeviationBound(Bounds.LOWER);
+            }
+            if ("upper".equals(bound)) {
+                return getStdDeviationBound(Bounds.UPPER);
+            }
+        }
+        return super.getProperty(path);
```

Comment:
The disadvantage of doing the change here is that we will have two ways of getting the upper and lower bounds. In buckets path syntax for pipeline aggregations we will have `std_deviation_bounds.upper` and `std_deviation_bounds.lower` and for the AggregationPath syntax used by the terms aggregation order parameter we will still use `std_upper` and `std_lower`. I now wonder if we should instead change the name in the `value(String name)` method so that it is `standard_deviation_bounds.upper` and `standard_deviation_bounds.lower`? Then it would be consistent all the places it is used. We could make the change in 6.0 and have a change in 5.x which accepts both versions but logs to the deprecation logger if the `std_upper` or `std_lower` version is used.

@clintongormley @jpountz what do you think?